src/modules/exceptions.ts

```diff
@@ -15,14 +15,23 @@ const MAX_STACK_FRAMES = 50;
  * detects whether each frame is user code (in_app: true) or library code (in_app: false).
  *
  * @param error - The error to capture (can be Error, string, object, or any value)
+ * @param contextStack - Optional Error object to use for stack context (for validation errors)
  * @returns ErrorData object with structured error information
  */
-export function captureException(error: unknown): ErrorData {
+export function captureException(
+  error: unknown,
+  contextStack?: Error,
+): ErrorData {
+  // Handle CallToolResult objects (SDK 1.21.0+ converts errors to these)
+  if (isCallToolResult(error)) {
+    return captureCallToolResultError(error, contextStack);
+  }
+
   // Handle non-Error objects
   if (!(error instanceof Error)) {
     return {
       message: stringifyNonError(error),
-      type: "NonError",
+      type: "UnknownErrorType",
       platform: "javascript",
     };
```

Comment:
I think based on their code, we know it is an McpError type I'm pretty sure. You could do a quick includes check or something here.